keras_nlp/metrics/bleu.py

```diff
@@ -357,11 +357,7 @@ def validate_and_fix_rank(inputs, tensor_name, base_rank=0):
             possible_matches,
             translation_length,
             reference_length,
-        ) = tf.py_function(
-            func=self._calculate_bleu_score,
-            inp=[y_true, y_pred],
-            Tout=[self.dtype, self.dtype, self.dtype, self.dtype, self.dtype],
-        )
+        ) = self._calculate_bleu_score(y_true, y_pred)
```

Comment:
This will break in-graph computation for TF, right? I suppose that's okay because this is an uncommon use case? Just figured we should verify that nobody is using this if that's the case.

One thing we could do is run it in `py_function` iff we're in the TF backend, but maybe that's too hacky for what we get from it

Reply:
Yeah, I was more interested in coming up with a cohesive story here that works on all backends. And honestly I don't think compiling bleu is really a use case.

- You cannot use this as a live metric during training, because no model will output an entire sequence suitable for bleu.
- You could specifically wrap this in a tf function after you have called `generate()` on a bunch of translations, summaries etc, but why bother? If our only option is wrapping in `py_function`, that will probably be slower than pure eager execution anyway.

Saves us time, and is clearer UX to have the same eager constraint everywhere.